### PR TITLE
[#25] Avoid unnecessary force pushes

### LINKED ISSUE #25: Avoid unnecessarily force-pushing update branch
`update-daemon` will always force-push the `automatic-update` branch even if there were no changes since the last update to this branch. Each such excessive force-push produces a notification which becomes quite annoying.
Ideally, `update-daemon` should only update the branch if something has changed since the last update.

### PR DESCRIPTION
**Pull request by lierdakil**
Problem: `setup_update_branch` unconditionally resets HEAD to the
default branch's HEAD before updating flake.lock. Consequently, even if
the new flake.lock is the same as the one currently on the update
branch, it will force-push regardless.

Solution: During `setup_update_branch`, force the update branch to point
to either the remote update branch head, or the default branch if remote
update branch does not exist. Ensure working tree is checked out to the
update branch.

If there are changes to the `flake.lock` compared to the update branch,
soft-reset to the default branch and commit the changes. Otherwise,
don't.

This uses the same force-checkout workflow as `init_repo`, hence the
code is factored out into a separate function. Simpler approaches (like
`checkout_head` etc) didn't work.

Also fix clippy complaints, run rustfmt and clean-up `flake.nix`.

Off-topic: `nix flake check` complains about `system.stateVersion`. Is it safe to ignore?

Disclaimer: I've tested this a bit, and it seems to work. However, the workflow is surprisingly convoluted, so I'm not 100% positive I've caught all the edge cases.

Resolves #25

### CODE REVIEW

**Review by rvem — APPROVED** [review]

File: src/git.rs [hunk 1]
```diff
@@ -176,21 +178,33 @@ pub fn setup_update_branch(
...
+        )
+        .map_err(SetupUpdateBranchError::FindDefaultBranch)?;
+
     let branch = if let Ok(b) = update_branch {
         let update_branch_commit = b
             .get()
             .peel_to_commit()
             .map_err(SetupUpdateBranchError::PeelUpdateBranchCommit)?;
-        if update_branch_commit.author().email() != Some(&settings.author.email) {
+        let default_branch_commit = b
```
Comment (rvem) [inline src/git.rs:193]: Shouldn't it be `default_branch.get().peel_to_commit()`? :thinking:

Comment (rvem) [reply src/git.rs:193]: AFAIU, currently `update_branch_commit.id() == default_branch_commit.id()` is always true

Comment (lierdakil) [reply src/git.rs:193]: Okay, sorry, *pushing* a fixup

**Review by rvem — APPROVED** [review]

**Comment by lierdakil** [comment]
FML, github merged the version of this branch with a fixup

**Comment by lierdakil** [comment]
What the actual fuck

**Pull request merged**